### DISCUSSION TITLE
Gradients (from-to) broken in 3.3.2 but fine in 3.2.7

### DISCUSSION THREAD

**Discussion by damieng** [tailwindlabs/tailwindcss#11193] [Help]
I have a site I maintain that is built using Nuxt 2 and was using Tailwind 2 in JIT mode. I upgraded to Tailwind 3.0.x a while ago going through the upgrade guide and all was fine. Now however upgrading from 3.2.7 breaks all my gradients including simple ones like `bg-gradient-to-l from-teal-500 to-cyan-500` - the from/to in the CSS seem to have broken start/stops:

![image](https://github.com/tailwindlabs/tailwindcss/assets/118951/1296d0ef-dfa0-4c16-b080-beea6bf936b5)

Here's a few things I've checked:

1. Running latest versions of all the packages 
2. Revisited all steps on the [nuxt 2 integration page](https://tailwindcss.com/docs/guides/nuxtjs) including config files
3. Components are not using conditional/string appending class names
4. Components with these gradients should be matched e.g. `/components/Header.vue`

It works just fine in nuxt's dev mode - it fails however when doing the generate/start and on prod deployments. 

Likewise if I set my tailwindcss to 3.2.7 it just works. Set it to 3.3.2 and it breaks.

Relevant fragments:

nuxt.config.js

```js 
  build: {
    postcss: {
      postcssOptions: {
        plugins: {
          tailwindcss: {},
          autoprefixer: {},
        },
      },
    },
    html: {
      minify: {
        removeRedundantAttributes: false,
      },
    },
  }
```

tailwind.config.js
```js
module.exports = {
  content: [
    "./components/*.{js,vue,ts}",
    "./layouts/**/*.vue",
    "./pages/**/*.vue",
    "./plugins/**/*.{js,ts}",
    "./nuxt.config.{js,ts}",
  ],
  theme: {
    fontSize: {
      xs: [
        "0.75rem",
        {
          lineHeight: "1.25rem",
        },
      ],
      sm: [
        "0.875rem",
        {
          lineHeight: "1.5rem",
        },
      ],
      base: [
        "1rem",
        {
          lineHeight: "1.75rem",
        },
      ],
      lg: [
        "1.125rem",
        {
          lineHeight: "2rem",
        },
      ],
      xl: [
        "1.25rem",
        {
          lineHeight: "1.75rem",
        },
      ],
      "2xl": [
        "1.5rem",
        {
          lineHeight: "2rem",
        },
      ],
      "3xl": [
        "1.875rem",
        {
          lineHeight: "2.25rem",
        },
      ],
      "4xl": [
        "2.25rem",
        {
          lineHeight: "2.75rem",
        },
      ],
      "5xl": [
        "3rem",
        {
          lineHeight: "1.05",
        },
      ],
      "6xl": [
        "3.75rem",
        {
          lineHeight: "1.05",
        },
      ],
      "7xl": [
        "4.5rem",
        {
          lineHeight: "1.05",
        },
      ],
      "8xl": [
        "6rem",
        {
          lineHeight: "1.05",
        },
      ],
      "9xl": [
        "8rem",
        {
          lineHeight: "1.05",
        },
      ],
    },
    extend: {
      fontFamily: {
        sans: ["Merriweather Sans", "sans-serif"],
      },
      spacing: {
        128: "28rem",
        160: "32rem",
      },
    },
  },
  plugins: [
    require("@tailwindcss/aspect-ratio"),
    require("@tailwindcss/forms"),
    require("@tailwindcss/typography"),
  ],
};
```

components/Header.vue

```html
<template>
  <header class="sticky z-50 top-0 bg-gradient-to-l from-teal-400 to-cyan-600">
   ...
  </header>
</template>
```
I can't see this issue reported anywhere so it's almost certainly some sort of combination of things/odd config issue but I can't figure it out.

**Comment by wongjn** [comment]
Could be something related to #10989 that was fixed by #11002 in [v3.3.2](https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.3.2).

**Reply by damieng** [reply]
Updated the discussion to indicate it's still a problem on 3.3.2.

Not sure where the default gradient positions are supposed to be coming from but looks like they're not happening in this nuxt 2 workflow somehow.

**Reply by StefanVesper** [reply]
any news here? Just ran into the same problem....

**Comment by jwkicklighter** [comment]
I am having the same issue using Nuxt 2. Downgrading to v3.2.7 fixed the problem. Any newer version (with v3.3.3 being the latest right now) and it works with dev mode but does not work in production (using `nuxt generate` and `nuxt start` to serve the site in static mode).

**Reply by dvalent** [reply]
☝ also my current solution. using Nuxt (2.15) Vue (2.7.10) and deploying on Netlify

**Comment by adamwathan** [comment]
If someone can provide a reproduction I’m happy to look, just don’t have time to set everything up from scratch myself.

The defaults should be getting added as empty CSS variables in the base layer along with our other empty variables.

**Comment by foxted** [comment]
The issue seem to be coming from Nuxt 2 default config, where `build.extractCSS` is set to false by default. Setting `build.extractCSS` to true in the config seems to fix the issue: 

```js
build: {
   extractCSS: true,
}
```

**Reply by dohahelmy** [reply]
Thank you for the good solution!!
I didn't want to downgrade and this works like a charm
It also had an effect on the missing classes assigned using `rtl:` and `ltr:` which works fine on development but not for production
